### PR TITLE
upload avatar

### PR DESCRIPTION
**Pull request by ThuyenPVT**
allow user upload their avatar : https://gyazo.com/b33fc2bdfb61753e27c100d106d916bc

### CODE REVIEW

File: lib/bloc/blocs/register_bloc.dart [hunk 1]
```diff
@@ -32,12 +36,14 @@ class RegisterBloc extends Bloc<RegisterEvent, RegisterState> {
...
     @required String password,
     @required String name,
-    String imageURL,
+    File imageURL,
   }) async* {
     yield RegisterState.loading();
     try {
       final String uid = await _userRepository.signUp(email, password);
-      await _accountRepository.createUser(uid, name, email, imageURL);
+      var _result = _storageRepository.uploadUserImage(uid, imageURL);
```
Comment (chuongdevJC) [inline lib/bloc/blocs/register_bloc.dart]: local variable dont need to define like _result -> result

File: lib/bloc/blocs/register_bloc.dart [hunk 2]
```diff
@@ -32,12 +36,14 @@ class RegisterBloc extends Bloc<RegisterEvent, RegisterState> {
...
+      var _imageURL = await (await _result).ref.getDownloadURL();
```
Comment (chuongdevJC) [inline lib/bloc/blocs/register_bloc.dart]: .

File: lib/data/source/remote/storage_remote_datasource.dart [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import 'dart:io';
+
+import 'package:firebase_storage/firebase_storage.dart';
+
+abstract class StorageRemoteDataSource {
+  Future<StorageTaskSnapshot> uploadUserImage(String _uid, File _image);
```
Comment (chuongdevJC) [inline lib/data/source/remote/storage_remote_datasource.dart]: change params name

File: lib/repositories/storage_repository.dart [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import 'package:firebase_storage/firebase_storage.dart';
+import 'package:injectable/injectable.dart';
+import 'package:structure_flutter/data/source/remote/storage_remote_datasource.dart';
+import 'package:structure_flutter/di/injection.dart';
+
+@singleton
+class StorageRepository {
+  final _storageRemoteDataSource = getIt<StorageRemoteDataSource>();
+
+  Future<StorageTaskSnapshot> uploadUserImage(String _uid, File _image) {
```
Comment (chuongdevJC) [inline lib/repositories/storage_repository.dart]: change params name

File: lib/bloc/blocs/register_bloc.dart (hunk 2 above)
Comment (ThuyenPVT) [reply lib/bloc/blocs/register_bloc.dart]: i use _imageURL to distinguish between imageURL param and local _imageURL !

**Review by truyenjc — APPROVED** [review]
LGTM!

**Review by chuongdevJC — APPROVED** [review]
LGTM!! 👍

**Pull request merged**